### PR TITLE
Moving language-agnostic components into a separate subproject

### PR DESCRIPTION
**Pull request by darora**
@gracew figure I'd get this out of the way first

No functionality changes included in this one

### CODE REVIEW

File: gradle/publish-runnable.gradle [hunk 1]
```diff
@@ -34,7 +34,7 @@ bintray {
     publish = true
     pkg {
         repo = 'releases'
-        name = 'groovy-language-server'
+        name = project.ext.get("applicationName")
```
Comment (darora) [inline gradle/publish-runnable.gradle:37]: is this the right way to be doing this?

File: language-server-commons/src/main/resources/log4j.properties [hunk 2]
```diff
@@ -0,0 +1,10 @@
+log4j.rootLogger=DEBUG, STDOUT, file
```
Comment (gracew) [inline language-server-commons/src/main/resources/log4j.properties:1]: why was this file added?

File: gradle/publish-runnable.gradle (hunk 1 above)
Comment (jiahuijiang) [inline gradle/publish-runnable.gradle:37]: for `project.ext` I think you need to actually declare the extra property inside each build.gradle.
`project.name` should work

**Comment by jiahuijiang** [comment]
@darora added you as collaborator so you can change the repository name too

File: language-server-commons/build.gradle [hunk 3]
```diff
@@ -0,0 +1,20 @@
+apply from: "${rootDir}/gradle/java.gradle"
+apply from: "${rootDir}/gradle/immutables-processors.gradle"
+
+sourceCompatibility = '1.8'
```
Comment (gracew) [inline language-server-commons/build.gradle:4]: you actually don't need this b/c it's in `gradle/java.gradle`

File: language-server-commons/build.gradle [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+dependencies {
+    compile "com.fasterxml.jackson.core:jackson-databind:${jacksonDatabindVersion}"
```
Comment (gracew) [inline language-server-commons/build.gradle:7]: delete, this is unused

File: language-server-commons/build.gradle [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    compile "com.google.guava:guava:${guavaVersion}"
+    compile "org.apache.commons:commons-io:${apacheCommonsVersion}"
+    compile "org.slf4j:slf4j-api:${slf4jApiVersion}"
+    compile "io.typefox.lsapi:io.typefox.lsapi:${typefoxLsApiVersion}"
+    compile "io.typefox.lsapi:io.typefox.lsapi.annotations:${typefoxLsApiVersion}"
+    compile "io.typefox.lsapi:io.typefox.lsapi.services:${typefoxLsApiVersion}"
+
+    testCompile "com.google.code.findbugs:annotations:${findbugsAnnotationsVersion}"
```
Comment (gracew) [inline language-server-commons/build.gradle:15]: delete

File: gradle/publish-runnable.gradle (hunk 1 above)
Comment (gracew) [reply gradle/publish-runnable.gradle:37]: +1, let's just use `project.name` here

If you want to use `ext` in the future (to specify additional properties), in this file you would need:

```
ext {
    applicationName
}
```

and then in the subproject build.gradle, add:

```
ext.applicationName = 'myname'
```

**Review by gracew — COMMENTED** [review]
apart from some minor comments this looks good to me!

**Comment by natgabb** [comment]
You're putting all the language server implementations in one repo? Why not separate repos?

The README.md needs to be updated.

**Comment by gracew** [comment]
I think the overhead associated with separate repos is too high for now.

**Comment by gracew** [comment]
(I don't want to merge a minor change to language-server-commons, tag a release, wait for build, then send 2 PRs to upgrade in groovy-language-server and sql-language-server)

**Comment by gracew** [comment]
@darora , you'll update the README once you add more of the sql scaffolding, right? :)

**Comment by natgabb** [comment]
@gracew fair enough :)

File: language-server-commons/src/main/resources/log4j.properties (hunk 2 above)
Comment (natgabb) [reply language-server-commons/src/main/resources/log4j.properties:1]: It's needed to log to file with log4j since the default is to stdout

**Comment by darora** [comment]
Cool, thanks, I'm going to clean this up in the morning and merge

**Comment by darora** [comment]
Closing in favour of #130
